connectors/couchbase/translator-couchbase/src/test/java/org/teiid/translator/couchbase/TestN1QLVisitor.java

```diff
@@ -0,0 +1,492 @@
+        helpTest(sql, N1QL0904);
+        
+        sql = "SELECT couchbase.TITLE(attr_string) FROM T2";
+        helpTest(sql, N1QL0905);
+        
+        sql = "SELECT couchbase.LTRIM(attr_string, 'This') FROM T2";
+        helpTest(sql, N1QL0906);
+        
+        sql = "SELECT couchbase.TRIM(attr_string, 'is') FROM T2";
+        helpTest(sql, N1QL0907);
+        
+        sql = "SELECT couchbase.RTRIM(attr_string, 'value') FROM T2";
+        helpTest(sql, N1QL0908);
+        
+        sql = "SELECT couchbase.POSITION(attr_string, 'is') FROM T2";
+        helpTest(sql, N1QL0909);
+    }
+    
+    @Test
+    public void testNumbericFunctions() throws TranslatorException {
+        
+        String sql = "SELECT CEILING(attr_double) FROM T2";
+        helpTest(sql, N1QL1001); 
+        
+        sql = "SELECT LOG(attr_double) FROM T2";
+        helpTest(sql, N1QL1002); 
+        
+        sql = "SELECT LOG10(attr_double) FROM T2";
+        helpTest(sql, N1QL1003); 
+        
+        sql = "SELECT RAND(attr_integer) FROM T2";
+        helpTest(sql, N1QL1004); 
+    }
+    
+    @Test
+    public void testConversionFunctions() throws TranslatorException {
+
+        String sql = "SELECT convert(attr_long, string) FROM T2";
+        helpTest(sql, N1QL1101);
+    }
+    
+    @Test
+    public void testDateFunctions() throws TranslatorException {
+        
+        String sql = "SELECT couchbase.CLOCK_MILLIS() FROM T2";
+        helpTest(sql, N1QL1201); 
+        
+        sql = "SELECT couchbase.CLOCK_STR() FROM T2";
+        helpTest(sql, N1QL1202); 
+        
+        sql = "SELECT couchbase.CLOCK_STR('2006-01-02') FROM T2";
+        helpTest(sql, N1QL1203);
+                
+        sql = "SELECT couchbase.DATE_ADD_MILLIS(1488873653696, 2, 'century') FROM T2";
+        helpTest(sql, N1QL1204); 
+        
+        sql = "SELECT couchbase.DATE_ADD_STR('2017-03-08', 2, 'century') FROM T2";
+        helpTest(sql, N1QL1205); 
+    }
+    
+    @Test
+    public void testProcedures() throws TranslatorException {
+       
+        String sql = "call getTextDocuments('%e%', 'test')";
+        helpTest(sql, N1QL1301);
+        
+        sql = "call getDocuments('customer', 'test')";
+        helpTest(sql, N1QL1302);
+        
+        sql = "call getTextDocument('customer', 'test')";
+        helpTest(sql, N1QL1303);
+        
+        sql = "call getDocument('customer', 'test')";
+        helpTest(sql, N1QL1304);
+        
+        sql = "call saveDocument('k001', 'test', '{\"key\": \"value\"}')";
+        helpTest(sql, N1QL1305);
+        
+        sql = "call deleteDocument('k001', 'test')";
+        helpTest(sql, N1QL1306);
+        
+        sql = "call getTextMetadataDocument('test')";
+        helpTest(sql, N1QL1307);
+        
+        sql = "call getMetadataDocument('test')";
+        helpTest(sql, N1QL1308);
+    }
+    
+    public static enum N1QL {
+        N1QL0101,
+        N1QL0102,
```

Comment:
> Could you not simply inline the expected results?
This is just for test usage, for make the unit test a little tidy, cause, the N1QL statement may very large, if you look at the N1QL.properties, the length of some expected result are 1000, if we put it inline, the test method looks awful.

> Did you need to replace any variables?
If REPLACE_EXPECTED set to true, all expected results in N1QL.properties will be replace, which same like test integration.